### PR TITLE
added and customized Book Controller,

### PR DESCRIPTION
**Pull request by JaysonJuza**
added BookDto and CreateBookRequestDto objects for the Book entity and for use at the service level, added the mapstruct library and added the corresponding dependency to pom.xml, added MapperConfig configuration class ,
added BookMapper for converting entity to DTO object and vice versa, custom exception added

### CODE REVIEW

File: src/main/java/onlinebookstoreapp/controller/BookController.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+@RestController
+@RequestMapping("/books")
+public class BookController {
+    private final BookService bookService;
+
+    @Autowired
+    public BookController(BookService bookService) {
+        this.bookService = bookService;
+    }
```
Comment (MAMentorFX) [inline src/main/java/onlinebookstoreapp/controller/BookController.java]: Optional: consider @RequiredArgsConstructor.

File: src/main/java/onlinebookstoreapp/dto/BookDto.java [hunk 2]
```diff
@@ -0,0 +1,17 @@
+package onlinebookstoreapp.dto;
+
+import java.math.BigDecimal;
+import lombok.Getter;
+import lombok.Setter;
+
+@Getter
+@Setter
```
Comment (MAMentorFX) [inline src/main/java/onlinebookstoreapp/dto/BookDto.java]: It's better to use @Data for Dtos

File: src/main/java/onlinebookstoreapp/mapper/BookMapper.java [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+import onlinebookstoreapp.dto.BookDto;
+import onlinebookstoreapp.dto.CreateBookRequestDto;
+import onlinebookstoreapp.model.Book;
+import org.mapstruct.Mapper;
+
+@Mapper(config = MapperConfig.class)
+public interface BookMapper {
+    BookDto toDtoModel(Book book);
+
+    Book toBookModel(CreateBookRequestDto requestDto);
```
Comment (MAMentorFX) [inline src/main/java/onlinebookstoreapp/mapper/BookMapper.java]: Either toBookDto + toBook, or toDto + toModel

File: src/main/java/onlinebookstoreapp/service/impl/BookServiceImpl.java [hunk 4]
```diff
@@ -10,19 +14,31 @@
...
 public class BookServiceImpl implements BookService {
     private final BookRepository bookRepository;
+    private final BookMapper bookMapper;
 
     @Autowired
-    public BookServiceImpl(BookRepository bookRepository) {
+    public BookServiceImpl(BookRepository bookRepository, BookMapper bookMapper) {
         this.bookRepository = bookRepository;
+        this.bookMapper = bookMapper;
     }
```
Comment (MAMentorFX) [inline src/main/java/onlinebookstoreapp/service/impl/BookServiceImpl.java]: Speaking of @RequiredArgsConstructor: that's why you wont be bothered changing constructor in the future, when you'll be adding more and more fields.

File: src/main/java/onlinebookstoreapp/service/impl/BookServiceImpl.java [hunk 5]
```diff
@@ -10,19 +14,31 @@
...
+        return bookMapper.toDtoModel(bookRepository.save(book));
     }
 
     @Override
-    public List<Book> findAll() {
-        return bookRepository.findAll();
+    public List<BookDto> findAll() {
+        return bookRepository.findAll().stream()
+                .map(bookMapper::toDtoModel)
+                .toList();
```
Comment (MAMentorFX) [inline src/main/java/onlinebookstoreapp/service/impl/BookServiceImpl.java]: MapStruct supports mapping lists

**Review by MAMentorFX — CHANGES_REQUESTED** [review]

**Review by okuzan — APPROVED** [review]

**Pull request merged**